distribution/std-lib/Base/src/Data/Maybe.enso

```diff
@@ -0,0 +1,46 @@
+from Base import all
+
+## A type representing computations that may fail.
+type Maybe
+
+    ## No contained value.
+    Nothing
+
+    ## A value.
+    type Just value
```

Comment:
Are we sure `Just` is a good name for this? I think scala's `Some` is much clearer in its intent – `Just value` always felt to me more like a pun than a type name.